influent-java/src/main/java/influent/forward/MsgPackPingDecoder.java

```diff
@@ -0,0 +1,122 @@
+package influent.forward;
+
+import org.msgpack.value.ImmutableArrayValue;
+import org.msgpack.value.ImmutableValue;
+import org.msgpack.value.Value;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.Optional;
+
+final class MsgPackPingDecoder {
+  private static final Logger logger = LoggerFactory.getLogger(MsgPackPingDecoder.class);
+
+  private final ForwardSecurity security;
+  private final Optional<ForwardClientNode> node;
```

Comment:
I heard that we should use Optional for return values.
http://javazuki.com/articles/optional-introduction.html

I feel odd, but this is Java style……